plugins/obs-filters/cross-stitching-filter.c

```diff
@@ -0,0 +1,259 @@
+	gs_eparam_t                    *greyscale_param;
+
+	uint8_t                         stitch_gap;
+	uint8_t                         negative_bias;
+	struct vec4                     stitch_color;
+	bool                            invert;
+	bool                            greyscale;
+};
+
+
+/*
+ * As the functions' namesake, this provides the user facing name
+ * of your Filter.
+ */
+static const char *cross_stitching_filter_name(void *unused)
+{
+	UNUSED_PARAMETER(unused);
+	return obs_module_text("Cross.Stitching");
+}
+
+/*
+ * This function is called (see bottom of this file for more details
+ * whenever the OBS filter interface changes. So when the user is messing
+ * with a slider this function is called to update the internal settings
+ * in OBS, and hence the settings being passed to the CPU/GPU.
+ */
+static void cross_stitching_filter_update(void *data, obs_data_t *settings)
+{
+	struct cross_stitching_filter_data *filter = data;
+	filter->stitch_gap = (uint8_t)obs_data_get_int(settings,
+		SETTING_STITCH_GAP);
+	filter->negative_bias = (uint8_t)obs_data_get_int(settings,
+		SETTING_NEGATIVE_BIAS);
+	uint32_t stitch_color = (uint32_t)obs_data_get_int(settings,
+		SETTING_STITCH_COLOR);
+	uint32_t stitch_opacity = (uint32_t)obs_data_get_int(settings,
+		SETTING_OPACITY);
+	filter->invert = obs_data_get_bool(settings, SETTING_INVERT);
+	filter->greyscale = obs_data_get_bool(settings, SETTING_GREYSCALE);
+
+	/* Convert Stitch Color from a uint32_t to a vec4 */
+	stitch_color &= 0xFFFFFF;
+	stitch_color |= ((stitch_opacity * 255) / 100) << 24;
+	vec4_from_rgba(&filter->stitch_color, stitch_color);
+}
+
+/*
+ * Since this is C we have to be careful when destroying/removing items from
+ * OBS. Jim has added several useful functions to help keep memory leaks to
+ * a minimum, and handle the destruction and construction of these filters.
+ */
+static void cross_stitching_filter_destroy(void *data)
+{
+	struct cross_stitching_filter_data *filter = data;
+
+	if (filter->effect) {
+		obs_enter_graphics();
+		gs_effect_destroy(filter->effect);
+		obs_leave_graphics();
+	}
+
+	bfree(data);
+}
+
+/*
+ * When you apply a filter OBS creates it, and adds it to the source. OBS also
+ * starts rendering it immediately. This function doesn't just 'create' the
+ * filter, it also calls the render function (farther below) that contains the
+ * actual rendering code.
+ */
+static void *cross_stitching_filter_create(obs_data_t *settings, obs_source_t *context)
+{
+	/*
+	 * Because of limitations of pre-C99 compilers, you can't create an
+	 * array that doesn't have a know size at compile time. The below
+	 * function calculates the size needed and allocates memory to
+	 * handle the source.
+	 */
+	struct cross_stitching_filter_data *filter =
+		bzalloc(sizeof(struct cross_stitching_filter_data));
+
+	/*
+	 * By default the effect file is stored in the ./data directory that
+	 * your filter resides in.
+	 */
+	#ifdef WIN32
```

Comment:
Sorry. I had edited my response with the updated reply from Jim, but your comment hadn't shown up yet for me, hence the disturbance in the force.